backend/src/controllers/register.controller.ts

```diff
@@ -0,0 +1,155 @@
+import { Router, Request, Response } from 'express';
+import RegisterService from '../services/register.service';
+import UserEntity from '../entities/user.entity';
+import bcrypt from 'bcrypt';
+
+// Função de validação para o CEP
+const validateCep = (cep: string): boolean => {
+  const cepRegex = /^[0-9]{5}-?[0-9]{3}$/;
+  return cepRegex.test(cep);
+};
+
+// Função de validação para o e-mail
+const validateEmail = (email: string): boolean => {
+  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+  return emailRegex.test(email);
+};
+
+// Função de validação para a senha
+const validatePassword = (password: string): boolean => {
+  return password.length >= 6; // mínimo de 6 caracteres
+};
+
+class RegisterController {
+  private prefix: string = '/register';
+  public router: Router;
+  private registerService: RegisterService;
+
+  constructor(router: Router, registerService: RegisterService) {
+    this.router = router;
+    this.registerService = registerService;
+    this.initRoutes();
+  }
+
+  private initRoutes() {
+    this.router.post(this.prefix, (req: Request, res: Response) => this.createUser(req, res));
+    this.router.get(this.prefix, (req: Request, res: Response) => this.getUsers(req, res));
+    this.router.get(`${this.prefix}/:id`, (req: Request, res: Response) => this.getUser(req, res));
+    this.router.put(`${this.prefix}/:id`, (req: Request, res: Response) => this.updateUser(req, res));
+    this.router.delete(`${this.prefix}/:id`, (req: Request, res: Response) => this.deleteUser(req, res));
+  }
+
+  private async createUser(req: Request, res: Response) {
+    const { name, email, gender, paymentMethod, cpf, cep, password, recoveryQuestion } = req.body;
+
+    if (!name || !email || !gender || !paymentMethod || !cpf || !cep || !password || !recoveryQuestion) {
```

Comment:
This line could be a verification method as like validateCep, validateEmail and validatePassword